### ISSUE TITLE
Missing accessible name on post card on timelines

### ISSUE THREAD

**Issue by biibirll** [cheeaun/phanpy#734]
## Bug description

On all timelines, when navigating by keyboard shortcuts (J and K) with a screen reader on (e.g., NVDA), the screen reader reports only a link with no text, which is the first identifiable element of each post's card.

- Site: phanpy.social
- Site version: 2024.08.31.1b8ab47
- Instance: bolha.us

## To Reproduce

1. Go to https://phanpy.social
2. With NVDA on, enter focus mode by pressing NVDA + Space.
3. Jump some posts by pressing J
4. Jump back a few posts by pressing K
5. Notice that NVDA only announces "link"

## Expected behavior

The screen reader should announce the post content in a way that is meaningful, like it happens on Mastodon's default web app, including user who posted, post content, media descriptions and other related information.

## Desktop
- OS: Windows 11 
- Browser: Brave
- Version: 1.69.160 Chromium: 128.0.6613.114

## Additional context

- NVDA version: 2024.3
